File: BE/robot_delivery/src/main/java/com/example/robot_delivery/model/request/CreateOrderRequest.java
package com.example.robot_delivery.model.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CreateOrderRequest {
    private String recipientPhone;
    private Double startLat;
    private Double startLng;
    private Double deliveryLat;
    private Double deliveryLng;
}

File: Mobile/robot_delivery/lib/app/data/models/request/create_order_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';
part 'create_order_request.freezed.dart';
part 'create_order_request.g.dart';

@freezed
abstract class CreateOrderRequest with _$CreateOrderRequest {
  const factory CreateOrderRequest({
    required String recipientPhone,
    required double startLat,
    required double startLng,
    required double deliveryLat,
    required double deliveryLng,
    String? senderAddress,
    String? deliveryAddress,
  }) = _CreateOrderRequest;



  factory CreateOrderRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateOrderRequestFromJson(json);
}